File: examples/baker-example/src/main/java/webshop/simple/ShipItemsInstance.java
package webshop.simple;

import webshop.simple.ingredients.ReservedItems;
import webshop.simple.ingredients.ShippingAddress;

public class ShipItemsInstance implements ShipItems {
    @Override
    public ShippingConfirmed apply(ShippingAddress shippingAddress, ReservedItems reservedItems) {
        return new ShippingConfirmed();
    }
}

File: examples/baker-example/src/main/scala/webshop/webservice/recipe/ShipItemsInstance.scala
package webshop.webservice.recipe

import cats.effect.IO
import cats.effect.unsafe.implicits.global
import webshop.webservice.recipe.CheckoutFlowEvents.ShippingConfirmed
import webshop.webservice.recipe.CheckoutFlowIngredients.{ReservedItems, ShippingAddress}
import webshop.webservice.recipe.CheckoutFlowInteractions.ShipItems

import scala.concurrent.Future
import scala.concurrent.duration._

class ShipItemsInstance() extends ShipItems {

  override def apply(shippingAddress: ShippingAddress, reservedItems: ReservedItems): Future[ShippingConfirmed] = {
      IO.sleep(500.millis)
        .as(ShippingConfirmed())
        .unsafeToFuture()
  }
}

